core/src/main/java/org/apache/calcite/jdbc/CalciteMetaImpl.java

```diff
@@ -608,6 +608,8 @@ public MetaResultSet getTableTypes(ConnectionHandle ch) {
             new ArrayList<List<Object>>());
     boolean done = fetchMaxRowCount == 0 || rows.size() < fetchMaxRowCount;
     @SuppressWarnings("unchecked") List<Object> rows1 = (List<Object>) rows;
+    // Set ResultSet to null for supporting multi queries in same statement.
+    stmt.setResultSet(null);
```

Comment:
This is too aggressive :(

The iterator is not released after a call ended because it might not be fully read. If I'm not wrong, setting it to null would cause infinite loop when result row count is large enough.

Reply:
Thanks for your review and sorry for lately reply @zhztheplayer , updated the PR to address your comment.